wpimath/src/main/java/edu/wpi/first/math/kinematics/DifferentialDriveOdometry.java

```diff
@@ -63,12 +99,31 @@ public DifferentialDriveOdometry(Rotation2d gyroAngle) {
    * @param gyroAngle The angle reported by the gyroscope.
    */
   public void resetPosition(Pose2d poseMeters, Rotation2d gyroAngle) {
+    resetPosition(poseMeters, gyroAngle, 0, 0);
+  }
+
+  /**
+   * Resets the robot's position on the field.
+   *
+   * <p>The gyroscope angle does not need to be reset here on the user's robot code. The library
+   * automatically takes care of offsetting the gyro angle.
+   *
+   * @param poseMeters The position on the field that your robot is at.
+   * @param gyroAngle The angle reported by the gyroscope.
+   * @param leftDistanceMeters The distance traveled by the left encoder.
+   * @param rightDistanceMeters The distance traveled by the right encoder.
+   */
+  public void resetPosition(
+      Pose2d poseMeters,
+      Rotation2d gyroAngle,
+      double leftDistanceMeters,
+      double rightDistanceMeters) {
```

Comment:
This argument order doesn't match the constructor.

Reply:
I think it should stay separate since the reset() requires a pose argument but the constructor does not. We could resolve this by removing the reset(pose, gyroAngle) version, which I would be in favor of removing for all pose estimators / odometry.

Reply:
This constructor takes a pose argument: https://github.com/wpilibsuite/allwpilib/pull/4645/files#diff-1d8cdcaaa987b6a5636d3014013aacb3dac8a24f4763cb3cd2f63d107b6084e3R40
```java
  public DifferentialDriveOdometry(
      Rotation2d gyroAngle,
      double leftDistanceMeters,
      double rightDistanceMeters,
      Pose2d initialPoseMeters) {
```

Reply:
I will remove the functions which assume default encoder readings then.